xdsl/traits.py

```diff
@@ -342,3 +344,18 @@ def get_callable_region(cls, op: Operation) -> Region:
         Returns the body of the operation
         """
         raise NotImplementedError
+
+
+@dataclass(frozen=True)
+class HasCanonicalisationPatternsTrait(OpTrait):
+    """
+    Provides the rewrite passes to canonicalize an operation.
+    """
+
+    def verify(self, op: Operation) -> None:
+        return
+
+    @classmethod
+    @abc.abstractmethod
```

Comment:
Why do we have an implementation if we have an abstract method?
I would replace it with ...